### ISSUE TITLE
bert embedding make OOM in albert

### ISSUE THREAD

**Issue by urekalion** [huggingface/transformers#4569]
# ❓ Questions & Help

## Details
hi, i made item collaborative filtering model by albert model.
i set vocab size to 600k, it made OOM. 
(after initialized model, memory used 3 GB. but During initialization, it takes 11 GB)

i tracked source code and i found this: 
albert model initialize albert word embedding layer after initializing bert word embedding layer.
embedding layer is initialized twice. so it made OOM. 
- bert word embedding : vocab size x hidden layer size(4048)
- albert word embedding : vocab size x embedding size(128) => bert word embedding free

is there any problem if i fix AlbertEmbeddings to nn.Module?

thanks

`
class AlbertEmbeddings(BertEmbeddings):
    """
    Construct the embeddings from word, position and token_type embeddings.
    """

    def __init__(self, config):
        super().__init__(config)

        self.word_embeddings = nn.Embedding(config.vocab_size, config.embedding_size, padding_idx=config.pad_token_id)
        self.position_embeddings = nn.Embedding(config.max_position_embeddings, config.embedding_size)
        self.token_type_embeddings = nn.Embedding(config.type_vocab_size, config.embedding_size)
        self.LayerNorm = torch.nn.LayerNorm(config.embedding_size, eps=config.layer_norm_eps)

`
`
class BertEmbeddings(nn.Module):
    """Construct the embeddings from word, position and token_type embeddings.
    """

    def __init__(self, config):
        super().__init__()
        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id)
        self.position_embeddings = nn.Embedding(config.max_position_embeddings, config.hidden_size)
        self.token_type_embeddings = nn.Embedding(config.type_vocab_size, config.hidden_size)

        # self.LayerNorm is not snake-cased to stick with TensorFlow model variable name and be able to load
        # any TensorFlow checkpoint file
        self.LayerNorm = BertLayerNorm(config.hidden_size, eps=config.layer_norm_eps)
        self.dropout = nn.Dropout(config.hidden_dropout_prob)
`
